resolwe_bio/processes/WGBS/bsmap.yml

```diff
@@ -0,0 +1,119 @@
+# ================
+# Aligner - bsmap
+# ================
+---
+
+- slug: bsmap
+  name: BSMAP
+  requirements:
+    expression-engine: jinja
+  data_name: "bsmap ({{ reads|sample_name|default('?') }})"
+  version: 0.0.1
+  type: data:alignment:bam:bsmap
+  category: analyses:wgbs
+  flow_collection: sample
+  persistence: CACHED
+  description: >
+    BSMAP is a short reads mapping program for bisulfite sequencing in DNA methylation study.
+  input:
+    - name: genome
+      label: Reference genome
+      type: data:genome:fasta
+    - name: reads
+      label: Reads
+      type: data:reads:fastq
+    - name: mismatch_rate
+      label: Mismatch rate
+      type: basic:decimal
+      default: 0.08
+      description: >
+        If this value is between 0 and 1, it's interpreted as the mismatch rate w.r.t to the read
+        length. Otherwise it's interpreted as the maximum number of mismatches allowed on a read,
+        <=15.
+    - name: threads
+      label: Use this many processor threads
+      type: basic:integer
+      default: 8
+      description: >
+        Number of processor threads to be used when mapping reads to the reference sequence.
+    - name: rrbs
+      label: RRBS mapping mod
+      type: basic:string
+      required: False
+      description: >
+        Activating RRBS mapping mode and set restriction enzyme digestion sites.
+        Digestion position marked by '-', example: -D C-CGG for MspI digestion.
+    - name: mapping_strand
+      label: Mapping strand
+      type: basic:integer
+      default: 0
+      choices:
+        - label: Map to 2 forward strands
+          value: 0
+        - label: Map SE or PE reads to all 4 strands
+          value: 1
+      description: >
+        Set mapping strand information.
+  output:
+    - name: bam
```

Comment:
Objects of type data:alignment:bam should also have stats field in the outputs. Check if samtools flagstat produces something meaningful.